### ISSUE TITLE
Double check how we reflow for a huge height delta

### ISSUE THREAD

**Issue by zadjii-msft** [microsoft/terminal#16033]
Yes, exactly that. I did it for this test specifically:
https://github.com/microsoft/terminal/blob/fc4a37ee9189f2f7335ab942d50381a38a03acf2/src/buffer/out/ut_textbuffer/ReflowTests.cpp#L294-L315

`$` is the cursor. `false` in each line means that it has an explicit newline (`true` = force wrap).
It has a lot of comments about bugs in the old implementation, which is why the entire text past the cursor fits into the new buffer. In the new, fixed implementation the `false` = explicit newlines are correctly handled and so it doesn't fit anymore:
https://github.com/microsoft/terminal/blob/6374aece11edbf8b1329d56ab963e6ea2089f83d/src/buffer/out/ut_textbuffer/ReflowTests.cpp#L294-L315

Basically I had to make a choice over whether I:
* continue copying until all text in the old buffer has been consumed.
  If the new cursor position is now somewhere at a negative Y, we just put it at (0,0).
* stop copying at some point to prevent the cursor from being lost.

I chose the latter. What do you think we should do?

_Originally posted by @lhecker in https://github.com/microsoft/terminal/pull/15701#discussion_r1332947703_
